core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -70,6 +72,8 @@
 @LuceneTestCase.SuppressFileSystems("ExtrasFS")
 public class TranslogTests extends ElasticsearchTestCase {
 
+    private static final Pattern PARSE_LEGACY_ID_PATTERN = Pattern.compile("^" + Translog.TRANSLOG_FILE_PREFIX + "(\\d+)((\\.recovering))?$");
```

Comment:
yeah I moved it out since we really only use it in one context and I wanna discurage the usage